### ISSUE TITLE
Running Jupiter Lab getting Permission to listen on port xxxx denied.

### ISSUE THREAD

**Issue by MaximoTrinidad** [dotnet/interactive#605]
### Just executing the command: "jupyter lab" from "Anaconda PowerShell Prompt(Anaconda3)".

Please provide as much information as you can.
I don't know how this happened. It was working fine for some time and now it doesn't.

(base) PS [9] > jupyter lab
[W 19:42:37.290 LabApp] Permission to listen on port 8888 denied.
[W 19:42:37.291 LabApp] Permission to listen on port 8889 denied.
[W 19:42:37.291 LabApp] Permission to listen on port 8890 denied.
[W 19:42:37.292 LabApp] Permission to listen on port 8891 denied.
[W 19:42:37.293 LabApp] Permission to listen on port 8892 denied.
[W 19:42:37.293 LabApp] Permission to listen on port 8935 denied.
[W 19:42:37.294 LabApp] Permission to listen on port 8904 denied.
[W 19:42:37.294 LabApp] Permission to listen on port 8867 denied.
[W 19:42:37.295 LabApp] Permission to listen on port 8990 denied.
[W 19:42:37.295 LabApp] Permission to listen on port 8874 denied.
[W 19:42:37.296 LabApp] Permission to listen on port 8877 denied.
[W 19:42:37.296 LabApp] Permission to listen on port 8937 denied.
[W 19:42:37.297 LabApp] Permission to listen on port 8976 denied.
[W 19:42:37.297 LabApp] Permission to listen on port 8973 denied.
[W 19:42:37.298 LabApp] Permission to listen on port 8925 denied.
[W 19:42:37.299 LabApp] Permission to listen on port 8831 denied.
[W 19:42:37.300 LabApp] Permission to listen on port 8892 denied.
[W 19:42:37.300 LabApp] Permission to listen on port 8854 denied.
[W 19:42:37.301 LabApp] Permission to listen on port 8960 denied.
[W 19:42:37.302 LabApp] Permission to listen on port 8913 denied.
[W 19:42:37.303 LabApp] Permission to listen on port 8870 denied.
[W 19:42:37.304 LabApp] Permission to listen on port 8792 denied.
[W 19:42:37.305 LabApp] Permission to listen on port 8814 denied.
[W 19:42:37.305 LabApp] Permission to listen on port 8863 denied.
[W 19:42:37.306 LabApp] Permission to listen on port 8879 denied.
[W 19:42:37.307 LabApp] Permission to listen on port 8948 denied.
[W 19:42:37.308 LabApp] Permission to listen on port 8954 denied.
[W 19:42:37.309 LabApp] Permission to listen on port 8983 denied.
[W 19:42:37.309 LabApp] Permission to listen on port 8892 denied.
[W 19:42:37.310 LabApp] Permission to listen on port 8891 denied.
[W 19:42:37.311 LabApp] Permission to listen on port 8798 denied.
[W 19:42:37.312 LabApp] Permission to listen on port 8846 denied.
[W 19:42:37.313 LabApp] Permission to listen on port 8853 denied.
[W 19:42:37.316 LabApp] Permission to listen on port 8922 denied.
[W 19:42:37.317 LabApp] Permission to listen on port 8880 denied.
[W 19:42:37.319 LabApp] Permission to listen on port 8987 denied.
[W 19:42:37.319 LabApp] Permission to listen on port 8854 denied.
[W 19:42:37.320 LabApp] Permission to listen on port 8807 denied.
[W 19:42:37.321 LabApp] Permission to listen on port 8985 denied.
[W 19:42:37.322 LabApp] Permission to listen on port 8861 denied.
[W 19:42:37.322 LabApp] Permission to listen on port 8912 denied.
[W 19:42:37.323 LabApp] Permission to listen on port 8870 denied.
[W 19:42:37.324 LabApp] Permission to listen on port 8843 denied.
[W 19:42:37.325 LabApp] Permission to listen on port 8871 denied.
[W 19:42:37.326 LabApp] Permission to listen on port 8962 denied.
[W 19:42:37.326 LabApp] Permission to listen on port 8903 denied.
[W 19:42:37.327 LabApp] Permission to listen on port 8987 denied.
[W 19:42:37.328 LabApp] Permission to listen on port 8955 denied.
[W 19:42:37.329 LabApp] Permission to listen on port 8940 denied.
[W 19:42:37.331 LabApp] Permission to listen on port 8867 denied.
[W 19:42:37.332 LabApp] Permission to listen on port 8980 denied.
[C 19:42:37.333 LabApp] ERROR: the notebook server could not be started because no available port could be found.
(base) PS [62] >

Any idea on How-To fix this issue? 

Thanks

### Please complete the following:

**Which version of .NET Interactive are you using?** There are a few ways to find this out:

* In a notebook, run the `#!about` magic command. 
N/A

* At the command line, run `dotnet interactive --version`.
(base) PS [63] > dotnet interactive --version
1.0.135102+2c83d127c2f9ba9ef179572fb928e12f29cb5afa

* In VS Code, run "Report installed version for .NET Interactive` and copy the version number from the status popup.
N/A
![JupyterLabPortDenied_01_2020-07-01_19-45-34](https://user-images.githubusercontent.com/19892435/86301301-8873b780-bbd3-11ea-82ac-fc4ee8cb4430.jpg)

Using Anaconda with Python 3.7:
(base) PS [124] > conda --version
conda 4.8.2

- OS
    - [X] Windows 10 version 2004
    - [ ] macOS
    - [ ] Linux (Please specify distro)
    - [ ] iOS
    - [ ] Android
 - Browser 
    - [X] Chrome
    - [ ] Edge
    - [ ] Firefox
    - [ ] Safari
 - Frontend
    - [ ] Jupyter Notebook
    - [X] Jupyter Lab - Anaconda 
    - [ ] nteract
    - [ ] Visual Studio Code
    - [ ] Other (please specify)

**Comment by jonsequitur** [comment]
It looks like `dotnet-interactive` doesn't have permission to open a port. We should fail gracefully when this happens. 

There are two workarounds:

If you're using JavaScript variable sharing, you'll need this port open. Adding a firewall rule might resolve it.

If you're not using JavaScript variable sharing, you can disable the feature in your `kernel.json` (which you can find by running `jupyter kernelspec list` in your Anaconda prompt):

```diff
{
  "argv": [
    "dotnet",
    "interactive",
    "jupyter",
    "--default-kernel",
    "csharp",
-   "--http-port-range",
-   "1000-3000",
    "{connection_file}"
  ],
  "display_name": ".NET (C#)",
  "language": "C#"
}
```

**Comment by MaximoTrinidad** [comment]
@jonsequitur 

I will try your workaround suggestion.  But, why this happened after been working for a long time in Windows 10?
Very strange!

:)

**Comment by MaximoTrinidad** [comment]
@jonsequitur 

No luck! None of the Kernel.json files have the "--http-port-range" setting.  I check no other VM's and they are working without it. 

:(

**Comment by MaximoTrinidad** [comment]
AAAHHHH!!

I've been fighting trying to get .NET Interactive Notebook to work, I then try to continue working with Docker containers.
Oh Boy! I got no ports available.  

After doing more research, I found some interesting posts leading to the possibility that the issue is related to the "Internet Connection Sharing" service.

The strange thing is that I can't disable, or stop, the service even if I try with the administrator user-id.  

My workaround: 

1. I'm able to give it a service user-id with a strong password.

2. Then, I went to the "Local Group Policy" editor,  and "Enabled" "Prohibit use of Internet Connection Sharing on your DNS domain network" rule under the "Computer Configuration | Administrative Template | Network | Network Connections".

Now, I got it working again! 

Able to work with both .NET Interactive Jupyter Notebook and my Docker containers.

But still, I have no idea what cause this problem!

I'm going to go ahead and close this issue.
